### PR TITLE
[12.0][MIG] project_taskcode

### PR DESCRIPTION
**Pull request by bouvyd**

### CODE REVIEW

File: project_task_code/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+{
+    "name": "Sequential Code for Tasks",
+    "version": "12.0.1.1.0",
+    "category": "Project Management",
+    "author": "OdooMRP team, "
+              "AvanzOSC, "
+              "Tecnativa, "
+              "Odoo Community Association (OCA)",
+    "website": "http://www.avanzosc.es",
```
Comment (nikul-serpentcs) [inline project_task_code/__manifest__.py:12]: @dbo-odoo IMO Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

File: project_task_code/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Copyright 2016 Tecnativa <vicent.cubells@tecnativa.com>
```
Comment (nikul-serpentcs) [inline project_task_code/models/__init__.py:1]: Remove Copyright from all `__init__`

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: project_task_code/__manifest__.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 {
     "name": "Sequential Code for Tasks",
-    "version": "11.0.1.1.0",
+    "version": "12.0.1.1.0",
```
Comment (pedrobaeza) [inline project_task_code/__manifest__.py:6]: We restart the numbers on each version, so it should be 12.0.1.0.0

File: project_task_code/models/project_task.py [hunk 4]
```diff
@@ -15,20 +15,19 @@ class ProjectTask(models.Model):
          _('The code must be unique!')),
     ]
 
-    @api.model
-    def create(self, vals):
-        if vals.get('code', '/') == '/':
-            vals['code'] = self.env['ir.sequence'].next_by_code('project.task')
-        return super(ProjectTask, self).create(vals)
+    @api.model_create_multi
```
Comment (pedrobaeza) [inline project_task_code/models/project_task.py:18]: Thanks for converting this to batch create :smiley:

**Review by pedrobaeza — APPROVED** [review]

File: project_task_code/tests/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Copyright 2016 Tecnativa <vicent.cubells@tecnativa.com>
```
Comment (alexey-pelykh) [inline project_task_code/tests/__init__.py:1]: __init__ files should not contain Copyright

File: project_task_code/models/project_task.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    _sql_constraints = [
+        ('project_task_unique_code', 'UNIQUE (code)',
+         _('The code must be unique!')),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('code', '/') == '/':
+                vals['code'] = self.env['ir.sequence'].next_by_code('project.task')
```
Comment (alexey-pelykh) [inline project_task_code/models/project_task.py:22]: Line too long, please check lint issues

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by alexey-pelykh** [comment]
@dbo-odoo do you have any ETA on fixes?
@pedrobaeza in what timeframe PR is deemed unattended?

**Comment by pedrobaeza** [comment]
There's no timeframe specified, but this is an Odoo employee that made this PR in a voluntary effort inside Hacktoberfest initiative, so I would say that we can superseed the PR for finishing it.

**Comment by alexey-pelykh** [comment]
Sorry for being a bit impatient, yet there's a #453 that fixes this mentioned above

**Comment by pedrobaeza** [comment]
Superseeded by #453

**Pull request closed without merging**